Sources/BugsnagPerformance/Private/SpanAttributesProvider.h

```diff
@@ -14,14 +14,14 @@ class SpanAttributesProvider {
     SpanAttributesProvider() noexcept;
     ~SpanAttributesProvider() {};
     
-    NSDictionary *networkSpanUrlAttributes(NSURL *url, NSError *encounteredError) noexcept;
-    NSDictionary *networkSpanAttributes(NSURL *url, NSURLSessionTask *task, NSURLSessionTaskMetrics *metrics,
-                                        NSError *encounteredError) noexcept;
-    NSDictionary *appStartSpanAttributes(NSString *firstViewName, bool isColdLaunch) noexcept;
-    NSDictionary *appStartPhaseSpanAttributes(NSString *phase) noexcept;
-    NSDictionary *viewLoadSpanAttributes(NSString *className, BugsnagPerformanceViewType viewType) noexcept;
-    NSDictionary *preloadedViewLoadSpanAttributes(NSString *className, BugsnagPerformanceViewType viewType) noexcept;
-    NSDictionary *viewLoadPhaseSpanAttributes(NSString *className, NSString *phase) noexcept;
+    NSMutableDictionary *networkSpanUrlAttributes(NSURL *url, NSError *encounteredError) noexcept;
```

Comment:
Why not keep the result type as `NSDictionary` and have the caller then call `mutableCopy`?

Reply:
We're trying to minimize allocator churn by breaking immutability rules internally in some controlled cases (keeping it mutable across the whole usage chain rather than re-allocating it as mutable).